octodns_azure/__init__.py

```diff
@@ -128,7 +131,12 @@ def __init__(
         self.relative_record_set_name = record.name or '@'
         self.record_type = record._type
         self._record = record
-        self.traffic_manager = traffic_manager
+        # new version of traffic manager throws the following unless you use the
+        # SubResource object for the target_resource
+        # AttributeError: 'Profile' object has no attribute 'validate'
+        self.traffic_manager = (
+            SubResource(id=traffic_manager.id) if traffic_manager else None
+        )
```

Comment:
This seems like the sort of regression that should be fixed before it gets to a full release?

Reply:
Looks like the `SubResource` is the correct way to do it, the previous method isn't the correct one and just happened to work 🤷 https://github.com/Azure/azure-sdk-for-python/issues/30686#issuecomment-1583982112 I do wish this was better documented.

Reply:
Reverted this and bumped azure-mgmt-dns to 8.1.0b1 that doesn't have this regression. It's weird that devs just have to "know" the version matrix. We now also depend on one more beta package, both slated to have a stable release by end of the month.

Reply:
> It's weird that devs just have to "know" the version matrix

I've found that to be the case for both azure and google client libs in the past (google's breaks all the time b/c of changes in lower levels/protobufs, i think i've had to track down and file problems like this 2 or 3 times.)

I think ppl working on the public client libs at big companies don't actually use them for anything, or if they do it's in a monolith env where everything is ~latest.

Reply:
In my experience, there's usually a fair amount of "first party" versions of infrastructure as well as code/libraries that's largely used over third party public versions.